### DISCUSSION TITLE
Power derating

### DISCUSSION THREAD

**Discussion by mazi991** [NatLabRockies/floris#258] [Q&A] [answered]
Hello everyone,

I have the Cp and Ct data of an experimental wind turbine model for different incoming wind speeds at optimum pitch angle and optimum TSR, but only below rated. Does this affect FLORIS simulations? Is it necessary to have Cp and Ct curves for the full range of operating wind speeds (i.e. from cut-in to cut-out speed)?

Moreover, I would like to know how I can simulate different derating setpoints in FLORIS? 
Is it necessary to have different Cp and Ct curves for each desired derating setpoint?

Thank you in advance for your help.

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi Mazi991,

FLORIS will interpolate the turbine power and thrust coefficients over the wind speed. If you only have these curves defined for below-rated conditions, then FLORIS simulations will fail if the wind speed at any of the turbines falls outside of the performance curves range. Note that due to wake effects there may be turbines below cut-in wind speed in larger wind farms even when the ambient wind speed is higher than cut-in.

As far as I am aware, there is no straight-forward way to enable derating in FLORIS without calculating the performance curves for every derating setpoint. You may want to read Issue #213 since it is somewhat related. Practically, you may be able to overwrite the power coefficient interpolant function inside the turbine object to effectively change turbine derating setpoints. This would be practically identical to having a different turbine model for each turbine derating setpoint. You can do that by following the example in #223.
